rholang/src/test/scala/coop/rchain/rholang/interpreter/ReduceTest.scala

```diff
@@ -102,45 +122,49 @@ class ReduceSpec extends FlatSpec with Matchers {
         BitSet()
       )
 
-    val interpreter = Reduce.makeInterpreter
-    val sendTask    = interpreter.eval(send)(Env())
-    val receiveTask = interpreter.eval(receive)(Env())
-
     val inspectTaskSendFirst = for {
-      _      <- sendTask
-      _      <- receiveTask
-      tsHash <- interpreter.tupleSpace.spaceMVar.take
-      _      <- interpreter.tupleSpace.spaceMVar.put(HashMap())
+      tupleSpace  <- Task.pure(InMemoryStore.create[Channel, List[Channel], List[Quote], Par])
+      interpreter <- Task.pure(Reduce.makeInterpreter(tupleSpace))
+      _           <- interpreter.eval(send)(Env())
+      _           <- interpreter.eval(receive)(Env())
+      tsHash      <- Task.pure(tupleSpace.toHashMap)
     } yield tsHash
 
     val inspectTaskReceiveFirst = for {
-      _      <- receiveTask
-      _      <- sendTask
-      tsHash <- interpreter.tupleSpace.spaceMVar.take
-      _      <- interpreter.tupleSpace.spaceMVar.put(HashMap())
+      tupleSpace  <- Task.pure(InMemoryStore.create[Channel, List[Channel], List[Quote], Par])
+      interpreter <- Task.pure(Reduce.makeInterpreter(tupleSpace))
+      _           <- interpreter.eval(receive)(Env())
+      _           <- interpreter.eval(send)(Env())
+      tsHash      <- Task.pure(tupleSpace.toHashMap)
     } yield tsHash
 
```

Comment:
@henrytill Any Scala expertise on how I can recreate the InMemoryStore each time without having to create a new test?